posthog/settings.py

```diff
@@ -66,23 +66,31 @@ def print_warning(warning_lines: Sequence[str]):
 else:
     JS_URL = os.environ.get("JS_URL", "")
 
-SECURE_SSL_REDIRECT = False
+# This is set as a cross-domain cookie with a random value.
+# Its existence is used by the toolbar to see that we are logged in.
+TOOLBAR_COOKIE_NAME = "phtoolbar"
+
+# SSL & cookie defaults
+if os.environ.get("SECURE_COOKIES", None) is None:
+    # Default to True if in production
+    secure_cookies = not DEBUG and not TEST
+else:
+    secure_cookies = get_bool_from_env("SECURE_COOKIES", True)
+
+TOOLBAR_COOKIE_SECURE = secure_cookies
+SESSION_COOKIE_SECURE = secure_cookies
+CSRF_COOKIE_SECURE = secure_cookies
+SECURE_SSL_REDIRECT = secure_cookies
 
+# production mode
 if not DEBUG and not TEST:
-    SECURE_SSL_REDIRECT = True
-    SESSION_COOKIE_SECURE = True
     if os.environ.get("SENTRY_DSN"):
         sentry_sdk.init(
             dsn=os.environ["SENTRY_DSN"], integrations=[DjangoIntegration()], request_bodies="always",
         )
 
-if get_bool_from_env("LOCAL_HTTPS", False):
-    SECURE_SSL_REDIRECT = False
-    SESSION_COOKIE_SECURE = True
-
 if get_bool_from_env("DISABLE_SECURE_SSL_REDIRECT", False):
     SECURE_SSL_REDIRECT = False
-    SESSION_COOKIE_SECURE = False
```

Comment:
is this correct? https://posthog.com/docs/deployment/running-behind-proxy#infinite-redirect

Reply:
That's correct. Explicitly setting `SESSION_COOKIE_SECURE` to `False` caused problems with logging in for users who had the `DISABLE_SECURE_SSL_REDIRECT` set. They were still behind SSL, but the redirect was just done via nginx. Thus the cookies had to be secure. 

Worse, in the last version of settings.py, this `SESSION_COOKIE_SECURE = False` was set in a way that it couldn't be overridden.

Reply:
Great!